tests/bsim/bluetooth/host/gatt/device_name/src/common.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_TESTS_BSIM_BLUETOOTH_HOST_GATT_DEVICE_NAME_SRC_COMMON_H_
+#define ZEPHYR_TESTS_BSIM_BLUETOOTH_HOST_GATT_DEVICE_NAME_SRC_COMMON_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+#define ADVERTISER_NAME      "Advertiser Pro II"
+#define SERVER_ORIGINAL_NAME "Server Super Name"
+
+static char server_new_name[CONFIG_BT_DEVICE_NAME_MAX];
+
+static void generate_name(char *name, size_t length)
+{
+	for (int i = 0; i < length; i++) {
+		name[i] = (i % 26) + 97;
+	}
```

Comment:
```suggestion
	for (size_t i = 0; i < length; i++) {
		name[i] = (i % 26) + 97;
	}
```

How do we handle cases where `length` >= ARRAY_SIZE(server_new_name)`? 
Test case failure/assert?

Reply:
Will update the `size_t`.

> How do we handle cases where length >= ARRAY_SIZE(server_new_name)`?
Test case failure/assert?

Sorry I don't get it? It's not something that can happen in the test as it is right now, am I missing something?

Reply:
> Sorry I don't get it? It's not something that can happen in the test as it is right now, am I missing something?

It's not an issue in this PR, but when someone modifies or adds additional tests in the future, then this function may be called with a length value that is larger than the array. 

I suggest to make it fail loudly instead of risking overwriting some other RAM in that case, which is much harder to debug

Reply:
Oh I think I understand the misunderstanding. The function is not updating `server_new_name` directly, instead it takes a buffer and the length of it

Reply:
Oh... Yeah, my mind completely messed those 2 up :D Please ignore everything I've said :D